### ISSUE TITLE
SA1123: DoNotPlaceRegionsWithinElements

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#56]
http://www.stylecop.com/docs/SA1123.html

**Comment by pdelvo** [comment]
Ive written a code fix for this and im currently working on tests. If only the# endregion is ín the method body it does not report a diagnostic, if only the #region part is in the method it does. This seems inconsistent.

It currently does report a diagnostic for 
```
private static void Foo()
#region foo
{
    Foo();
}
#endregion
```

The region is not really in the method body. So should we report a diagnostic here or not?

**Comment by sharwell** [comment]
I understood this rule to only apply if the `#region` and `#endregion` tags are both *within* the body of an element. In the example above, the tags are outside the body.

Remember that we'll need to strictly separate SA1123 from [SA1124](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/57), so whatever rules are implemented should be easy to distinguish and explain.
